prow/flagutil/github.go

```diff
@@ -331,6 +332,29 @@ func (o *GitHubOptions) GitHubClientWithAccessToken(token string) (github.Client
 	return client, err
 }
 
+func (o *GitHubOptions) GitClientFactory(cookieFilePath string, cacheDir *string, dryRun bool) (gitv2.ClientFactory, error) {
+	var gitClientFactory gitv2.ClientFactory
+	if cookieFilePath != "" && o.TokenPath == "" && o.AppPrivateKeyPath == "" {
+		opts := gitv2.ClientFactoryOpts{
+			CookieFilePath: cookieFilePath,
+			CacheDirBase:   cacheDir,
+		}
+		var err error
+		gitClientFactory, err = gitv2.NewClientFactory(opts.Apply)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create git client from cookieFile: %v", err)
+		}
+	} else {
+		gitClient, err := o.GitClient(dryRun)
```

Comment:
this imo is a http resolver from git v2, but I'm not entirely sure, so leaving it to the next PR